### ISSUE TITLE
question about LSTM with batch normalization

### ISSUE THREAD

**Issue by swangb** [tensorflow/models#1101]
## Please let us know which model this issue is about (specify the top-level directory)
I am wondering whether Tensorflow has the module to implement multilayer LSTM with batch normalization?

I find class tf.contrib.rnn.LayerNormBasicLSTMCell, which looks more powerful, however, I have a problem to use it. I am trying to build a tf.contrib.rnn.MultiRNNCell with 2 layer LayerNormBasicLSTMCell. However, the output state is consists of 2 tuples.
(LSTMStateTuple(c=<tf.Tensor 'model/RNN/rnn/while/Exit_2:0' shape=(?, 512) dtype=float32>, h=<tf.Tensor 'model/RNN/rnn/while/Exit_3:0' shape=(?, 512) dtype=float32>), LSTMStateTuple(c=<tf.Tensor 'model/RNN/rnn/while/Exit_4:0' shape=(?, 512) dtype=float32>, h=<tf.Tensor 'model/RNN/rnn/while/Exit_5:0' shape=(?, 512) dtype=float32>))

If I expect a tensor representation of the state gotten from this:
 outputs, state = tf.nn.dynamic_rnn(stacked_lstm, model_input,
                                         sequence_length=num_frames,
                                         dtype=tf.float32)
What should I do with state (how to tanslate LSTMStateTuple(tuple) to LSTMStateTuple(tensor))?
In tensorflow/tensorflow/contrib/rnn/python/ops/core_rnn_cell_impl.py, it says that:

state_is_tuple: If True, accepted and returned states are 2-tuples of
        the `c_state` and `m_state`.  If False, they are concatenated
        along the column axis.  The latter behavior will soon be deprecated.
      activation: Activation function of the inner states.

However, the given option is only for single layer, how to concatenate with 2 layer?

Can anyone help me with that? Thanks

My code is followed:

class LayerNormBasicLSTM(models.BaseModel):

  def create_model(self, model_input, vocab_size, num_frames, **unused_params):
    Args:
      model_input: A 'batch_size' x 'max_frames' x 'num_features' matrix of
                   input features.
      vocab_size: The number of classes in the dataset.
      num_frames: A vector of length 'batch' which indicates the number of
           frames for each video (before padding).

    Returns:
      A dictionary with a tensor containing the probability predictions of the
      model in the 'predictions' key. The dimensions of the tensor are
      'batch_size' x 'num_classes'.
    """
    lstm_size = FLAGS.lstm_cells
    number_of_layers = FLAGS.lstm_layers

    ## Batch normalize the input
    #__init__(num_units, forget_bias=1.0, input_size=None, activation=tf.tanh, layer_norm=True, norm_gain=1.0, norm_shift=0.0, dropout_keep_prob=1.0, dropout_prob_seed=None)
    stacked_lstm = tf.contrib.rnn.MultiRNNCell(
            [
                 tf.contrib.rnn.LayerNormBasicLSTMCell(
                    lstm_size, forget_bias=1.0)
                for _ in range(number_of_layers)
                ],
            state_is_tuple=True)

    loss = 0.0
    with tf.variable_scope("RNN"):
      outputs, state = tf.nn.dynamic_rnn(stacked_lstm, model_input,
                                         sequence_length=num_frames,
                                         dtype=tf.float32)
    
    # state is a tuple with cell state and hiden state
    # state: The final state.  If `cell.state_size` is an int, this
    #will be shaped `[batch_size, cell.state_size]`.  If it is a
    #`TensorShape`, this will be shaped `[batch_size] + cell.state_size`.
    #If it is a (possibly nested) tuple of ints or `TensorShape`, this will
    #be a tuple having the corresponding shapes.
    print state
    c1_state, h1_state = state[0]
    print c1_state.get_shape()
    print h1_state.get_shape()
    c2_state, h2_state = state[-1]
    print c2_state.get_shape()
    print h2_state.get_shape()
    new_state1 = array_ops.concat([c1_state, h1_state], 1)
    print new_state1
    new_state2 = array_ops.concat([c2_state, h2_state], 1)
    print new_state2

    aggregated_model = getattr(video_level_models,
                               FLAGS.video_level_classifier_model)
    return aggregated_model().create_model(
        model_input=new_state,
        vocab_size=vocab_size,
        **unused_params)

**Comment by tatatodd** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!
